File: api-service/src/main/java/org/qbynet/chat/repository/UserRepository.java
package org.qbynet.chat.repository;

import org.qbynet.chat.entity.User;
import org.springframework.data.domain.Pageable;
import org.springframework.data.mongodb.repository.ReactiveMongoRepository;
import org.springframework.stereotype.Repository;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

@Repository
public interface UserRepository extends ReactiveMongoRepository<User, String> {
    Mono<User> findByRemoteId(String remoteId);

    Mono<User> findByUsername(String username);

    Mono<Boolean> existsByUsername(String username);

    Flux<User> findAllByUsernameStartsWithIgnoreCase(String username, Pageable pageable);

    Flux<User> findAllByNicknameContainingIgnoreCase(String nickname, Pageable pageable);
}

File: src/main/kotlin/org/cubewhy/qbychat/repository/UserRepository.kt
package org.cubewhy.qbychat.repository

import org.cubewhy.qbychat.entity.User
import org.springframework.data.mongodb.repository.ReactiveMongoRepository
import org.springframework.stereotype.Repository
import reactor.core.publisher.Mono

@Repository
interface UserRepository : ReactiveMongoRepository<User, String> {
    fun findByUsername(username: String): Mono<User>
    fun existsByUsernameIgnoreCase(username: String): Mono<Boolean>
}